packages/react-sdk/src/components/BoardBar/pdf/utils.ts

```diff
@@ -31,7 +31,15 @@ export function canvas(element: CanvasElement): Content {
 }
 
 export function image(element: ImageElement, base64content: string): Content {
-  const dataURL = 'data:' + element.mimeType + ';base64,' + base64content;
+  // Cheap way to detect image type
+  const decoded = atob(base64content).trim();
+  let mimeType = 'image/png';
+  // Note: JFIF is the magic string in JPEG files.
+  if (decoded.includes('JFIF')) {
```

Comment:
This would scan a huge PNG or other files. `JFIF` should also be somewhere at be beginning, right? Can we improve here?

Reply:
I am actually not entirely sure how to do this better. There is stuff (which is not deterministic) before this :(

For svg's we can just domparse things but for png and jpg I am not sure. We can use signatures on it with base64 but those fall apart with zip or tar alike formats. Not sure if we ever will encounter this but these seemed more future proof than relying on magic signatures.

Reply:
I think we're talking past each other :D

It looks like, that `JFIF` is somewhere at the start of the file.

```sh
❯ xxd -l 100 Katzen\ Witze.jpg 
00000000: ffd8 ffe0 0010 4a46 4946 0001 0101 0048  ......JFIF.....H
00000010: 0048 0000 ffdb 0043 0006 0405 0605 0406  .H.....C........
```

`includes()` parses the file until the end to find it, I think. For large files this is quite expensive.